app/static/js/graphs/bubble-graph.js

```diff
@@ -143,15 +238,110 @@ function drawBubbleGraph(selector, data, replication) {
+                if( selected.includes(element.getAttribute("trait")) && (element.getAttribute("class").search(selected_.attr('value')) !== -1)){
+                    return false;}
+                else if( (all_controller=="yes") && (element.getAttribute("class").search("disabled") == -1) ){
+                    return false;}
+                else{ return true;}
+
+            });
+
+
+            for(i = 0; i < selectedBubbles.length; i++) {
+                var classAttr = $(selectedBubbles[i]).attr('class');
+
+                if (classAttr.search("opaque") == -1){
+                    classAttr= classAttr +" opaque";
+
+                    }
+                classAttr = [...new Set(classAttr.split(" "))].join(" ");
+
+                var cxAttr = $(selectedBubbles[i]).attr('cx');
+                var pubmedidAttr = $(selectedBubbles[i]).attr('pubmedid');
+                var authorAttr = $(selectedBubbles[i]).attr('author');
+                var accessionAttr = $(selectedBubbles[i]).attr('accession');
+                var nAttr = $(selectedBubbles[i]).attr('N');
+                var diseaseOrTraitAttr = $(selectedBubbles[i]).attr('DiseaseOrTrait');
+                var traitAttr = $(selectedBubbles[i]).attr('trait');
+                var cyAttr = yScale(nAttr);
+                var rAttr = sizeScale(nAttr);
+                var ancestryAttr=$(selectedBubbles[i]).attr('ancestrySelection');
+
+                bubbleDataGroup.append("circle")
+                    .attr("class", classAttr)
+                    .attr("onclick", "circleClick(evt);")
+                    .attr("onmouseover", "circleMouseOver(evt);")
+                    .attr("cx", cxAttr)
+                    .attr("cy", cyAttr)
+                    .attr("r", rAttr)
+                    .attr("pubmedid", pubmedidAttr)
+                    .attr("author", authorAttr)
+                    .attr("accession", accessionAttr)
+                    .attr("N", nAttr)
+                    .attr("DiseaseOrTrait", diseaseOrTraitAttr)
+                    .attr("trait", traitAttr)
+                    .attr("PatternSelection","Pattern")
+                    .attr("ancestrySelection",ancestryAttr);
+
+            }
+
+        }
+
+
+
     });
 
+
     $(selector).find(".filter select[name='trait']").change(function() {
 
         clearSelected();
+        var selectedBefore = $(`${selector} #bubbleData circle`).not( function(index, element) {
+                if( element.getAttribute("ancestrySelection").search("ANCESTRY") !== -1 ){ return false;}
+                else{ return true;}
+               });
+
+        $(selectedBefore).remove();
+
+        var selectedBefore_pattern = $(`${selector} #bubbleData circle`).not(function(index, element) {
+                if( element.getAttribute("PatternSelection").search("NO") == -1 ){ return false;}
+                else{ return true;}
+               });
+
+         $(selectedBefore_pattern).remove();

```

Comment:
Is this removing selections from ancestry/trait elements? Might want to put this code inside the `clearSelected` function or in a new function `clearSelectedTraits`.